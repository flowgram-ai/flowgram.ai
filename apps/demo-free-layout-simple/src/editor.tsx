import {
  EditorRenderer,
  FreeLayoutEditorProvider,
} from '@flowgram.ai/free-layout-editor';

import { NodeAddPanel } from './components/node-add-panel';
import { Tools } from './components/tools'
import { Minimap } from './components/minimap'
import { useEditorProps } from './hooks/use-editor-props'
import '@flowgram.ai/free-layout-editor/index.css';
import './index.css';

export const Editor = () => {
  const editorProps = useEditorProps()
  return (
    <FreeLayoutEditorProvider {...editorProps}>
      <div className="demo-free-container">
        <div className="demo-free-layout">
          <NodeAddPanel />
          <EditorRenderer className="demo-free-editor" />
        </div>
        <Tools />
        <Minimap />
      </div>
    </FreeLayoutEditorProvider>
  )
};
