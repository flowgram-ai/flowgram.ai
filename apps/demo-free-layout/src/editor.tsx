import { EditorRenderer, FreeLayoutEditorProvider } from '@flowgram.ai/free-layout-editor';

import { DemoTools } from './components/tools';
import '@flowgram.ai/free-layout-editor/index.css';
import './styles/index.css';
import { nodeRegistries } from './nodes';
import { initialData } from './initial-data';
import { useEditorProps } from './hooks';

export const Editor = () => {
  const editorProps = useEditorProps(initialData, nodeRegistries);
  return (
    <div className="doc-free-feature-overview">
      <FreeLayoutEditorProvider {...editorProps}>
        <div className="demo-container">
          <EditorRenderer className="demo-editor" />
        </div>
        <DemoTools />
      </FreeLayoutEditorProvider>
    </div>
  );
};
