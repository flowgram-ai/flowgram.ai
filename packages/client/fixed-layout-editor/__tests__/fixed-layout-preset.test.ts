import { describe, it, expect, beforeEach } from 'vitest';
import { FlowDocument, FlowNodeFormData } from '@flowgram.ai/editor';

import { baseWithDataMock, formMock } from '../__mocks__/flow.mock';
import { createContainer } from './create-container';

describe('fixed-layout-preset', () => {
  let flowDocument: FlowDocument;
  beforeEach(() => {
    const container = createContainer({});
    flowDocument = container.get(FlowDocument);
  });
  it('fromJSON and toJSON', () => {
    flowDocument.fromJSON(baseWithDataMock);
    expect(flowDocument.toJSON()).toEqual(baseWithDataMock);
  });
  it('nodeEngine(v2) toJSON', async () => {
    const container = createContainer({
      nodeEngine: {},
      nodeRegistries: [
        {
          type: 'noop',
          formMeta: {
            render: () => undefined,
          },
        },
      ],
    });
    flowDocument = container.get(FlowDocument);
    flowDocument.fromJSON(formMock);
    expect(flowDocument.toJSON()).toEqual(formMock);
    const { formModel } = flowDocument.getNode('noop_0').getData(FlowNodeFormData);
    expect(formModel.getFormItemByPath('title').value).toEqual('noop title');
    formModel.getFormItemByPath('title').value = 'noop title2';
    expect(flowDocument.toJSON()).toMatchSnapshot();
  });
});
