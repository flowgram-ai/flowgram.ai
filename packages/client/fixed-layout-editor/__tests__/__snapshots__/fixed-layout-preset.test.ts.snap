// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`fixed-layout-preset > nodeEngine(v2) toJSON 1`] = `
{
  "nodes": [
    {
      "blocks": [],
      "data": {
        "title": "noop title2",
      },
      "id": "noop_0",
      "type": "noop",
    },
  ],
}
`;
